PhpAmqpLib/Connection/AMQPStreamConnection.php

```diff
@@ -45,7 +45,7 @@ public function __construct(
         $ssl_protocol = null,
         ?AMQPConnectionConfig $config = null
     ) {
-        if (func_num_args() === 17 || ($ssl_protocol !== null && $ssl_protocol instanceof AMQPConnectionConfig === false)) {
+        if ($ssl_protocol !== null && $ssl_protocol instanceof AMQPConnectionConfig === false) {
```

Comment:
this might be confusing, conditionA && conditionB === false, what is precedence?

Reply:
as mentioned here https://www.php.net/manual/en/language.operators.precedence.php first "instanceof" called and then comparison.
so this is the same as $ssl_protocol !== null && ($ssl_protocol instanceof AMQPConnectionConfig) === false

so we trigger error only when $ssl_protocol is string or smth else, but not an AMQPConnectionConfig instance

Reply:
so You can use `!instanceof AMQPConnectionConfig` ?

Reply:
not exactly. we can use `!$ssl_protocol instanceof AMQPConnectionConfig` which is the same as `$ssl_protocol instanceof AMQPConnectionConfig === false`